ada_2_fhir-r4/zibs2020/payload/0.0/nl-core-Patient.xsl

```diff
@@ -0,0 +1,360 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+Copyright © Nictiz
+
+This program is free software; you can redistribute it and/or modify it under the terms of the
+GNU Lesser General Public License as published by the Free Software Foundation; either version
+2.1 of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+
+The full text of the license is available at http://www.gnu.org/copyleft/lesser.html
+-->
+
+<xsl:stylesheet exclude-result-prefixes="#all"
+    xmlns="http://hl7.org/fhir"
+    xmlns:util="urn:hl7:utilities" 
+    xmlns:f="http://hl7.org/fhir" 
+    xmlns:xd="http://www.oxygenxml.com/ns/doc/xsl"
+    xmlns:nf="http://www.nictiz.nl/functions" 
+    xmlns:nm="http://www.nictiz.nl/mappings"
+    xmlns:uuid="http://www.uuid.org"
+    xmlns:xs="http://www.w3.org/2001/XMLSchema" 
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform" 
+    version="2.0">
+    
+    <xsl:output method="xml" indent="yes"/>
+    <xsl:strip-space elements="*"/>
+    
+    <xsl:variable name="referById" select="false()"/>
+    
+    <xd:doc>
+        <xd:desc>Global variable that can be called anywhere in the process to access, include or reference any unique patient present in the input data</xd:desc>
+    </xd:doc>
+    <!-- Based on variable 'patients' in _zib2017.xsl -->
+    <xsl:template name="getAllPatients" as="element()*">
+        <xsl:param name="in" select="."/>
+        <!-- Originally, this grouping contained the 'nf:ada-pat-id' function, but because this function is not used outside this variable I edited it out. Maybe it should be removed from 2_fhir_fhir_include and added to _zib2017? -->
+        <xsl:variable name="identifier" select="(identificatienummer[@root = $oidBurgerservicenummer],identificatienummer[not(@root = $oidBurgerservicenummer)])[1]"/>
+        <xsl:for-each-group select="$in//patient[not(patient)][not(@datatype = 'reference')][.//(@value | @code | @nullFlavor)]" group-by="concat($identifier/@root, $identifier/@value)">
+            <!-- This for-each-group should be enough to sort patients by their official identifier, but $patients does a second group-by. Apparently there are enough use cases where no identificatienummer is present (or masked?), or dummy identificatienummers that exist in multiple patients. So I guess this second group-by is necessary? If a use case uses the same patient multiple times, it still gets filtered out -->
+            <!-- nf:getGroupingKeyPatient() is a function that concatenates a lot of Patient values together. It is not only used here, but also when generating the reference -->
```

Comment:
Volgens mij was de gedachte hier dat het in de praktijk (op het LSP) voorkomt dat dezelfde patiënt verschillen kan bevatten in verschillende bronsystemen, dus bv. dat het adres verschilt tussen twee bronsystemen. Dan krijg je twee verschillende instances voor dezelfde patient.